src/app/ClusterInfo.h

```diff
@@ -18,103 +18,72 @@
 
 #pragma once
 
-#include <app/AttributePathParams.h>
 #include <app/util/basic-types.h>
 #include <assert.h>
 #include <lib/core/Optional.h>
 
 namespace chip {
 namespace app {
-static constexpr AttributeId kRootAttributeId = 0xFFFFFFFF;
 
+/**
+ * ClusterInfo is the representation of an attribute path or an event path used by ReadHandler, ReadClient, WriteHandler,
+ * Report::Engine etc, it uses some invalid values for representing the wildcard values for its fields and  contains a mpNext field
+ * so it can be used as a linked list.
+ */
+// TODO: The cluster info should be separated into AttributeInfo and EventInfo.
+// Note: The change will happen after #11171 with a better linked list.
 struct ClusterInfo
 {
-    enum class Flags : uint8_t
-    {
-        kFieldIdValid   = 0x01,
-        kListIndexValid = 0x02,
-        kEventIdValid   = 0x03,
-    };
+    // Endpoint Id is a uint16 number, and should between 0 and 0xFFFE
+    static constexpr EndpointId kInvalidEndpointId = 0xFFFF;
+    // The ClusterId, AttributeId and EventId are MEIs,
+    // 0xFFFF is not a valid manufacturer code, thus 0xFFFF'FFFF is not a valid MEI
+    static constexpr ClusterId kInvalidClusterId     = 0xFFFF'FFFF;
+    static constexpr AttributeId kInvalidAttributeId = 0xFFFF'FFFF;
+    static constexpr EventId kInvalidEventId         = 0xFFFF'FFFF;
+    // ListIndex is a uint16 number, thus 0xFFFF is not a valid list index.
+    static constexpr ListIndex kInvalidListIndex = 0xFFFF;
```

Comment:
These constants should be private.  There's no reason for anything outside this class to use them directly, right?